pandas/core/tools/datetimes.py

```diff
@@ -434,9 +449,11 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                     yearfirst=yearfirst,
                     require_iso8601=require_iso8601
                 )
-
             if is_datetime64_dtype(result) and box:
                 result = DatetimeIndex(result, tz=tz, name=name)
+            # GH 6415
+            elif arg_is_series and utc:
```

Comment:
The reason why I am doing this is because a Series is converted to a numpy array for the date conversion, and `_convert_listlike` needs an indicator to return back a Series. 

The alternative is to have `_convert_listlike` return an array and have `to_datetime` do the conversion and localization. Which option is preferable?